powersimdata/design/transmission/substations.py

```diff
@@ -0,0 +1,35 @@
+import pandas as pd
+
+
+def calculate_substation_capacity(grid):
+    """For each substation in a grid, calculate the total substation transmission
+    capacity (in a transport model, ignoring power flow).
+
+    :param powersimdata.input.grid.Grid grid: a grid instance.
+    :return: (*pandas.Series*) -- index is substation IDs, value are total transmission
+        capacity (MW).
+    """
+    # Get new branch data frame with 'from_sub' and 'to_sub' columns
+    branch = grid.branch.assign(
+        from_sub=grid.branch.from_bus_id.map(grid.bus2sub.sub_id),
+        to_sub=grid.branch.to_bus_id.map(grid.bus2sub.sub_id),
+    )
+    # Calculate total substation capacity for matching 'from_sub' branches
+    from_capacities = pd.Series(
+        {
+            from_sub: branch.loc[branch_list].query("to_sub != from_sub").rateA.sum()
+            for from_sub, branch_list in branch.groupby("from_sub").groups.items()
+        }
+    )
+    # Calculate total substation capacity for matching 'to_sub' branches
+    to_capacities = pd.Series(
+        {
+            to_sub: branch.loc[branch_list].query("to_sub != from_sub").rateA.sum()
+            for to_sub, branch_list in branch.groupby("to_sub").groups.items()
+        }
+    )
+    # Reindex, sum and return
+    from_capacities = from_capacities.reindex(grid.sub.index).fillna(0)
+    to_capacities = to_capacities.reindex(grid.sub.index).fillna(0)
+    total_capacities = from_capacities + to_capacities
```

Comment:
I'm wondering whether we could simply do
```python
branch = branch[branch.from_sub != branch.to_sub]
from_cap = branch.groupby(["from_sub"]).sum()["rateA"]
to_cap = branch.groupby(["to_sub"]).sum()["rateA"]
total_cap = from_cap.combine(to_cap, lambda x, y: x+y, fill_value=0)
```

Reply:
This does not filter out the branches that have the same `from_sub` and `to_sub`, but the `combine` method looks intriguing. I haven't used that before.

Reply:
You are right! Much cleaner.

Reply:
Done.